CRM/Core/Payment/Stripe.php

```diff
@@ -817,6 +817,10 @@ public function doRecurPayment(&$params, $amount, $stripe_customer) {
         VALUES (%1, %2, %3, %4, %5, '{$this->_islive}')", $query_params);
     }
 
+    // update recur processor_id with subscriptionId
+    if ($subscription_id && $recuring_contribution_id) {
+      CRM_Core_DAO::setFieldValue('CRM_Contribute_DAO_ContributionRecur', $recuring_contribution_id, 'processor_id', $subscription_id);
```

Comment:
Can you use the Contributionrecur.create API here instead of accessing the DAO object directly?  It's less likely to break in the future if we use the API.

Reply:
@mattwire ,  used api instead of DAO function